libflux/core/src/scanner/scanner.rl

```diff
@@ -139,61 +141,60 @@
 
     # This is the scanner used when parsing a string expression.
     string_expr := |*
-        "${" => { tok = STRINGEXPR; fbreak; };
-        '"' => { tok = QUOTE; fbreak; };
-        (string_lit_char - "\"")+ => { tok = TEXT; fbreak; };
+        "${" => { tok = TokenType::STRINGEXPR; fbreak; };
+        '"' => { tok = TokenType::QUOTE; fbreak; };
+        (string_lit_char - "\"")+ => { tok = TokenType::TEXT; fbreak; };
     *|;
 }%%
 
 %% write data nofinal;
 
-int scan(
-    int mode,
-    const unsigned char **pp,
-    const unsigned char *data,
-    const unsigned char *pe,
-    const unsigned char *eof,
-    const unsigned char **last_newline,
-    unsigned int *cur_line,
-    unsigned int *token,
-    unsigned int *token_start,
-    unsigned int *token_start_line,
-    unsigned int *token_start_col,
-    unsigned int *token_end,
-    unsigned int *token_end_line,
-    unsigned int *token_end_col
-) {
-    int cs = flux_start;
-    switch (mode) {
-    case 0:
-        cs = flux_en_main;
-        break;
-    case 1:
-        cs = flux_en_main_with_regex;
-        break;
-    case 2:
-        cs = flux_en_string_expr;
-        break;
+pub fn scan(
+    data: &[u8],
+    mode: i32,
```

Comment:
Right, so here's what I mean: the equivalent of an `unsigned char` in C/C++ is `u8` in rust, though there's also `char` as (essentially) an alias that carries semantic meaning.